File: SampleProblem-2Sum/Solutions/SampleSolution.cpp
class Solution {
public:
    unordered_map<int , int> indices;

    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            indices[nums[i]] = i+1;
        }

        int a, b;
        vector<int> res(2);
        for (int i = 0; i < n; i++) {
            int b = target - nums[i];
            if (indices[b] && indices[b] != i+1) {
                return {i, indices[b] - 1};
            }
        }

        return {};
    }
};

File: SampleProblem-2Sum/Solutions/SampleSolution.py
class Solution:
    def twoSum(nums: list, target: int) -> list:
        indices = dict()

        n = len(nums)
        for i in range(n):
            indices[nums[i]] = i+1

        for i in range(n):
            a = target - nums[i]
            if indices.get(a, 0) and indices.get(a, 0) != i+1:
                return [i, indices[a] - 1]
        
        return []